realm/src/main/java/io/realm/Realm.java

```diff
@@ -168,6 +168,9 @@
     private final Map<Class<?>, Method> initTableMethods = new HashMap<Class<?>, Method>();
     private final Map<Class<?>, Method> insertOrUpdateMethods = new HashMap<Class<?>, Method>();
     private final Map<Class<?>, Constructor> generatedConstructors = new HashMap<Class<?>, Constructor>();
+
+    // Maps classes to the name of the proxied class. Examples: Dog.class -> Dog, DogRealmProxy -> Dog
```

Comment:
I store both. The Map goes from Class to String and it's purpose is to reduce the need for introspection.